controller/controller.go

```diff
@@ -106,13 +113,17 @@ func NewController(f ...ControllerOption) *Controller {
 		fn(opts)
 	}
 
+	allowTopologyLoops := false
+	if opts.allowTopologyLoops {
+		allowTopologyLoops = true
+	}
```

Comment:
Do we need this variable? I guess if you pass `opts.allowTopologyLoops` in the call to `newGatewayAPITopologyBuilder` it will have virtually the same effect, no?